credentials/src/main/kotlin/SSI.kt

```diff
@@ -270,41 +268,95 @@ public class VerifiablePresentation private constructor() {
+                      break
+                    }
+                  }
+                }
+              }
+
+              if (fieldMatch) {
+                selectableCredentials.add(vcJwt)
+              }
+            }
+          }
+        }
+      }
+
+      return selectableCredentials
+    }
   }
 }
 
+
 /**
- * The selectFrom method is a helper function that helps filter out the verifiable credentials which can not be selected and returns
- * the selectable credentials.
+ * Validates a list of Verifiable Credentials (VcJwts) against a presentation definition.
+ *
+ * This method checks the presentation definition and the list of Verifiable Credentials to ensure
+ * that the required fields in the definition are satisfied by the provided credentials.
+ *
+ * Currently, if the presentation definition contains submission requirements, or if any required field in an input
+ * descriptor is not satisfied by the provided credentials, an exception is thrown.
  *
- * @param presentationDefinition definition of what is expected in the presentation.
- * @param vcJwts verifiable credentials are the credentials from wallet provided to the library to find selectable credentials.
+ * ### Throws
+ * - [NotImplementedError] if the Presentation Definition contains Submission Requirements or if a Field Filter is implemented.
+ * - [Exception] if any required field is not satisfied in the given InputDescriptor.
  *
- * @return the selectable credentials.
+ * @param presentationDefinition The [PresentationDefinitionV2] object representing the presentation's definition.
+ * @param vcJwts The list of [VcJwt] representing the Verifiable Credentials to be validated against the presentation definition.
+ * @throws Exception If a required field is not satisfied in a given InputDescriptor.
  */
 @Throws(Exception::class)
-private fun selectFrom(presentationDefinition: PresentationDefinitionV2, vcJwts: List<VcJwt>): List<VcJwt> {
-  val selectableCredentials = mutableListOf<VcJwt>()
-
+private fun validatePresentationFrom(presentationDefinition: PresentationDefinitionV2, vcJwts: List<VcJwt>) {
   if (!presentationDefinition.submissionRequirements.isNullOrEmpty()) {
     throw NotImplementedError("Presentation Definition's Submission Requirements feature is not implemented")
   } else {
     for (inputDescriptor: InputDescriptorV2 in presentationDefinition.inputDescriptors) {
-      // Fields Processing
       if (inputDescriptor.constraints.fields!!.isNotEmpty()) {
-        for (vcJwt: VcJwt in vcJwts) {
-          var fieldMatch = false
-
-          for (field: FieldV2 in inputDescriptor.constraints.fields) {
-
-            // Optional fields are not needed to complete the required fields in the presentation definition
-            if (field.optional != null && field.optional) {
-              continue;
-            }
+        for (field: FieldV2 in inputDescriptor.constraints.fields) {
+          if (field.optional == true) continue // Skip optional fields
```

Comment:
(not urgent) are we to skip validation of optional fields if they do exist? what if an optional field does show up but it's the wrong format / type / pattern / etc?

Reply:
This function is basically seeing if the VCs you provided provide the bare minimum for the presentation definition.

If a PD needs firstName lastName and optional middleName

and you give it 4 creds:
firstName, LastName, MiddleName, address

it will satisfy the PD and ideally make a PR with only firstName lastName